jupyter/scripts/incremental_learning/experiment_driver.py

```diff
@@ -128,42 +132,109 @@ def read_dataset(_run, dataset_name):
 
 
 @ex.capture
-def transform_dataset(dataset: pd.DataFrame, transform_name: str, transform_parameters, _seed: int) -> pd.DataFrame:
+def transform_dataset(dataset: pd.DataFrame,
+                      test_fraction: float,
+                      transform_name: str,
+                      transform_parameters,
+                      _seed: int) -> pd.DataFrame:
+    if transform_name == 'partition_on_metric_ranges':
+        train_dataset, update_dataset = partition_on_metric_ranges(
+            dataset=dataset,
+            seed=_seed,
+            metric_features=transform_parameters['metric_features'])
+        train_dataset, test1_dataset = train_test_split(train_dataset, test_size=test_fraction)
+        update_dataset, test2_dataset = train_test_split(update_dataset, test_size=test_fraction)
+        return train_dataset, update_dataset, test1_dataset, test2_dataset
+
     if transform_name == 'resample_metric_features':
-        transformed_dataset = resample_metric_features(data_frame=dataset,
-                                                       seed=_seed, 
-                                                       fraction=transform_parameters['fraction'], 
-                                                       magnitude=transform_parameters['magnitude'],
-                                                       features=transform_parameters['features'])
-        return transformed_dataset
-    # TODO extend to other transforms
-    else:
-        raise NotImplementedError("This dataset transform is not integrated yet.")
+        train_dataset, test1_dataset = train_test_split(dataset, test_size=test_fraction)
+        update_dataset = resample_metric_features(
+            dataset=train_dataset,
+            seed=_seed, 
+            fraction=transform_parameters['fraction'], 
+            magnitude=transform_parameters['magnitude'],
+            metric_features=transform_parameters['metric_features'])
+        test2_dataset = resample_metric_features(
+            dataset=test1_dataset,
+            seed=_seed, 
+            fraction=transform_parameters['fraction'], 
+            magnitude=transform_parameters['magnitude'],
+            metric_features=transform_parameters['metric_features'])
+        return train_dataset, update_dataset, test1_dataset, test2_dataset
+
+    if transform_name == 'shift_metric_features':
+        train_dataset, test1_dataset = train_test_split(dataset, test_size=test_fraction)
+        update_dataset = shift_metric_features(
+            dataset=train_dataset,
+            seed=_seed,
+            fraction=transform_parameters['fraction'],
+            magnitude=transform_parameters['magnitude'],
+            categorical_features=transform_parameters['categorical_features'])
+        test2_dataset = shift_metric_features(
```

Comment:
There is a lot of code repetition here due to the "switch" statement like functionality. It's possible to use runtime reflection to do this more concisely, but this can be a future refactoring.

Reply:
I think streamlining this is a good idea. I'm going to stop short of generating the code, but I made it a lot less verbose in [e0b90ec](https://github.com/elastic/ml-cpp/pull/1994/commits/e0b90ec26570996b34ecf45218ead741a974e5cd).